test/13_arraycommon.jl

```diff
@@ -821,4 +821,19 @@ end
     @test z ≅ x
 end
 
+@testset "sort() fast sorting" begin
+    int_cate = rand(1:1000, 10_000) |> categorical
+    int_cate_missing = rand(vcat(missing,1:1000), 10_000) |> categorical
+    str_cate = rand([randstring(rand(1:32)) for i=1:1000], 10_000) |> categorical
+    str_cate_missing =  rand(vcat(missing,[randstring(rand(1:32)) for i=1:1000]), 10_000) |> categorical
+
+    for rev in [true, false], v in (int_cate, int_cate_missing, str_cate, str_cate_missing)
+        sa = sort(v, rev = rev)
+        @test issorted(sa, rev = rev)
```

Comment:
I realized that checking whether the result is sorted isn't enough to ensure it's correct. It would be good to also check `sa == sort!(convert(Array, v), ...)`. Also, a second series of checks should change the order of levels via `levels!` and check that `issorted` is still true.